beacon-chain/db/db.go

```diff
@@ -17,6 +18,10 @@ import (
 type BeaconDB struct {
 	db           *bolt.DB
 	DatabasePath string
+
+	// Beacon chain deposits in memory.
+	deposits     []*depositContainer
```

Comment:
instead of using a slice, wouldnt it be better to store deposits in a map ? We could use the hash of the deposit as the key when storing it in the map. The number of pending deposits will not be large enough to matter in the normal case but till chainstart we would be storing 16384 deposits in memory, this is where performance might be an issue

Reply:
The benefit of using a map would be that only unique values are stored. I'm not sure of the performance trade off yet, I'll have to compare map vs list iterations.

Reply:
Yeah its something we can keep in mind in the future, although shouldnt be an issue now for our testnet